src/test/munit/oracle/arrays-and-structs-xa-source-oracle-test-case.xml

```diff
@@ -0,0 +1,114 @@
+                <munit:parameters>
+                    <munit:parameter propertyName="config" value="oracleXADbPoolingConfig"/>
+                </munit:parameters>
+            </munit:parameterization>
+        </munit:parameterizations>
+    </munit:config>
+
+    <munit:before-test name="createATableForListenerTest">
+        <try>
+            <db:execute-script config-ref="${config}">
+                <db:sql><![CDATA[DROP TABLE  TEST_XA_TRANSACTION]]></db:sql>
+            </db:execute-script>
+            <error-handler>
+                <on-error-continue enableNotifications="true" logException="false">
+                </on-error-continue>
+            </error-handler>
+        </try>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[CREATE TABLE TEST_XA_TRANSACTION (id number)]]></db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql>create or replace type TEST_STRUCT as object (field1 varchar2(3), field2 number(2), field3 varchar2(60))</db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql>create or replace type TEST_STRUCT_COLLECTION as table of TEST_STRUCT</db:sql>
+        </db:execute-ddl>
+        <db:execute-ddl config-ref="${config}">
+            <db:sql><![CDATA[
+create or replace procedure retain_first_struct (z in out TEST_STRUCT_COLLECTION)
+    IS
+begin
+    if z is not null and z.count > 1 then
+        z := TEST_STRUCT_COLLECTION(z(1));
+    end if;
+end;
+]]></db:sql>
+        </db:execute-ddl>
+    </munit:before-test>
+
+    <munit:test name="prepareArrayAndStructSourceXATransaction" ignore="#[java!org::mule::extension::db::DbMunitUtils::isTestIgnored('oracle')]">
+        <munit:enable-flow-sources>
+            <munit:enable-flow-source value="dbListenerFlow"/>
+        </munit:enable-flow-sources>
+
+        <munit:execution>
+            <db:insert config-ref="${config}">
+                <db:sql><![CDATA[INSERT INTO TEST_XA_TRANSACTION(ID) VALUES(1)]]></db:sql>
+            </db:insert>
+
+            <munit-tools:dequeue queueName="sourceLocalTransactionTestQueue" timeout="100" timeoutUnit="SECONDS"/>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-equals actual="#[payload.list[0][1]]" expected="#[1]"/>
+        </munit:validation>
+    </munit:test>
+
+    <munit:after-suite name="dropStuff">
+        <try>
+            <db:execute-script config-ref="${config}">
+                <db:sql>
+                    drop procedure retain_first_struct;
+                    drop type TEST_STRUCT_COLLECTION;
+                    drop type TEST_STRUCT;
+                </db:sql>
+            </db:execute-script>
+            <error-handler>
+                <on-error-continue logException="false"/>
+            </error-handler>
+        </try>
+    </munit:after-suite>
+
+    <flow name="dbListenerFlow">
```

Comment:
@giss-ignacio This flow tests the happy path, could you add another test with a flow with an operation not belonging to the XA transaction, also using var.list and placed between the prepareArray and the store-procedure?

Reply:
@acurci-at-mulesoft According to the new information shared by Cristian Chan, that test cannot be done, right? so we can close this conversation and merge the pr?